src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Scanner;
+
+import duke.task.Task;
+
+/**
+ * Represents a storage class for I/O operations.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Returns a storage instance.
+     * @param filePath Path to the file duke.txt.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads file at filePath.
+     * @return Scanner sc
+     */
+    public Scanner load() {
+        try {
+            File f = new File(filePath);
+            Scanner sc = new Scanner(f);
+            return sc;
+        } catch (FileNotFoundException fileError) {
+            Ui.showError("Error in loading data. File not found.");
+            return new Scanner("");
+        }
+    }
+
+    /**
+     * Writes content to file at filePath.
+     * @param it Iterator of the arrayList from TaskList.
+     */
+    public void save(Iterator<Task> it) {
+        try {
+            FileWriter fw = new FileWriter(filePath, false);
+            while (it.hasNext()) {
+                fw.write(it.next().toStringSaveFormat());
+            }
+            fw.close();
+            Ui.showError("Successfully saved contents into duke.txt");
```

Comment:
Perhaps can use another method in Ui? Using showError sounds not corresponding to successfully saving contents.

Reply:
Thanks for pointing it out, might have accidentally used the wrong method 😅 .